documentation/src/docs/retention.md

```diff
@@ -0,0 +1,158 @@
+ - If `S` has a retention policy set, consider it for truncation.
+ - Generate a Tail Stream Cut, `TSC` near the tail end of the Stream where writers could still be appending data.
+ - Add `TSC` to the Retention Set for the Stream.
+ - Compute the current size of the Stream from Head Stream Cut `HSC` to `TSC`.
+ - Based on size of the Stream, arrive at a truncation Stream Cut using the following algorithm:
+   - If the size of `S` is less than min limit, then do nothing.
+   - If size of `S` is greater than min limit and max limit is not set, find a Stream Cut SC closest to but greater than the min limit, such that truncating at SC leaves more data in `S` than min limit.
+   - If a max limit is set, and the size of `S` is greater than the max limit, find a Stream Cut SC in the retention set closest to but smaller than the max limit, such that truncating at SC leaves less data than the max limit but more data than the min limit in S.
+   - If there is a max limit set and the size of `S` is between the max and the min limits, then attempt to truncate `S` at a Stream Cut SC closest to its min limit.
+
+     ![Stream truncation](img/Figure1.png)
+
+Stream S with Segments `S1` to `S12` and Stream Cuts `SC1`, `SC2`, `SC3`, `SC4`, `SC5`, `SC6`.
+
+### Consumption Based Retention
+
+#### Need
+
+Size and time based retention policies are suitable for applications that need to store data over the longer term (archival). 
+In some scenarios, however, streaming data is transient (short lived), or storage space is at a premium; consequently, reclaiming storage space as soon as applications no longer need the data is highly desirable in such scenarios.
+Use cases that require such space reclamation include but are not limited to the following:
+1. A message queue - When using a Pravega Stream as a message queue, events in the queue can be deleted post consumption by all subscribers.
+2. Deployments with limited storage capacity - The storage available on small-footprint environments like edge devices (gateways) is typically constrained. Once data is no longer needed, perhaps because it has moved out of the edge to a core data center or to the cloud, it can be deleted to create space for more incoming data.
+   
+#### Terminology
+- Subscriber Reader Group - A Subscriber Reader Group is a Reader Group whose read positions are considered for computing the truncation point for a Stream with Consumption Based Retention. When such a Subscribing Reader Group goes down, the system will truncate the Stream based on last published position of this Reader Group and will not truncate any data _not_ read by this Reader Group.
+- Non-Subscriber Reader Group - A Non-Subscriber Reader Group (default) is the one that wants to read from a Stream, but does not want the consumption to impact Stream truncation. When this Reader Group goes down and comes back up, it may have lost some messages in the queue because those were deleted while it was away.
```

Comment:
"lost some messages in the queue" -> "lost some messages in the Stream"

Reply:
fixed